testsupport/wait/host.go

```diff
@@ -1064,6 +1064,30 @@ func (a *HostAwaitility) WaitForNotifications(username, notificationType string,
 	return notifications, err
 }
 
+// WaitForNotificationWithName waits until there is an expected Notifications available for the provided user and with the notification type and which match the conditions (if provided).
+func (a *HostAwaitility) WaitForNotificationWithName(notificationName, notificationType string, criteria ...NotificationWaitCriterion) ([]toolchainv1alpha1.Notification, error) {
+	a.T.Logf("waiting for notification with name '%s'", notificationName)
+	var notifications []toolchainv1alpha1.Notification
+	err := wait.Poll(a.RetryInterval, a.Timeout, func() (done bool, err error) {
+		notification := &toolchainv1alpha1.Notification{}
+		if err := a.Client.Get(context.TODO(), types.NamespacedName{Name: notificationName, Namespace: a.Namespace}, notification); err != nil {
+			return false, err
+		}
+		if typeFound, found := notification.GetLabels()[toolchainv1alpha1.NotificationTypeLabelKey]; !found {
+			return false, fmt.Errorf("notification found with name does not have type label")
+		} else if typeFound != notificationType {
+			return false, fmt.Errorf("notification found with name does not have the expected type")
+		}
+		notifications = append(notifications, *notification)
```

Comment:
If we find a notification and append it here but the call to `matchNotificationWaitCriterion` returns false then `wait.Poll` will run again and we'd end up appending the notification multiple times in `notifications`? If we get rid of the slice then no need to worry about this.

Reply:
That's a good point! Thank you for pointing that out. I was returning a slice before because the functions `matchNotificationWaitCriterion` and `printNotificationWaitCriterionDiffs` take a slice as input parameter, and I was trying to minimize change. But as you pointed out, that is a good reason to create a new function which take single notification as input. 
done in [f571b95](https://github.com/codeready-toolchain/toolchain-e2e/pull/560/commits/f571b955373036242fc903480344d27435a26df1)